### ISSUE TITLE
ZAPE possibly wrong?

### ISSUE THREAD

**Issue by olivier-s-j** [lavinei/pybats_nbdev#1]
https://github.com/lavinei/pybats_nbdev/blob/9c7697c8476c42899346641f1c41184da9799aca/pybats_nbdev/loss_functions.py#L88

The Zape is defined as:

math:: ZAPE(y, f) = \\frac{1}{n} \sum_{i=1:n} I(y_i = 0) * f_i + I(y_i = 1) * |y_i-f_i| / y_i

But shouldn't it be:

math:: ZAPE(y, f) = \\frac{1}{n} \sum_{i=1:n} I(y_i = 0) * |f_i| + I(y_i = 1) * |(y_i-f_i) / y_i|

I.e.: 

`  y = np.ravel(y)
    f = np.ravel(f)
    nonzeros = y.nonzero()[0]
    n = len(y)
    loss = np.abs(np.copy(f))
    loss[nonzeros] = np.abs((y[nonzeros] - f[nonzeros])/ y[nonzeros]) 
    return 100*np.mean(loss)`

As the ZAPE is like the MAPE, but where the ground truth is 0, you fill in your forecast. 
Also, the APEs, are always positive.

FYI: the MAPE calculation in scikit-learn: https://github.com/scikit-learn/scikit-learn/blob/7a1aad0a3/sklearn/metrics/_regression.py#L197
